app/controllers/contractors_controller.rb

```diff
@@ -29,10 +29,25 @@ def update
     end
   end
 
+  def deactivate_account
+  end
+
+  def deactivate
+    if current_contractor && current_contractor.authenticate(params[:contractor][:password])
+      current_contractor.update_attribute(:status, "inactive")
+      session[:contractor_id] = nil
+      flash[:notice] = "Account successfully deactivated."
+      redirect_to root_path
+    else
+      flash.now[:error] = 'Invalid login data'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.